### PR TITLE
fix quota sync issues

### LINKED ISSUE #8858: core component failed to start due to quota migration failure.
```
Aug 28 06:34:59 192.168.112.1 core[2686]: 2019-08-28T06:34:59Z [INFO] [/core/main.go:100]: Start to sync quota data .....
Aug 28 06:34:59 192.168.112.1 core[2686]: 2019-08-28T06:34:59Z [ERROR] [/core/main.go:102]: Fail to sync quota data, failed to check health: Get http://chartmuseum:9999/health: dial tcp 192.168.160.4:9999: connect: connection refused
Aug 28 06:34:59 192.168.112.1 core[2686]: 2019-08-28T06:34:59Z [FATAL] [/core/main.go:242]: quota migration error, failed to check health: Get http://chartmuseum:9999/health: dial tcp 192.168.160.4:9999: connect: connection refused
```

Seems it should retry a few times before failing while checking the chartmuseum's health

### LINKED ISSUE #8859: Syncing quota data fails in large Harbor instance
I have a 1.8.x env that has 1.1 TB image data and thousands of tags.

When I'm running `1.8 -> 1.9` upgrade, the core component tries to sync the quota data but fails due to the following error:
```
Aug 28 06:35:01 192.168.112.1 core[2686]: 2019-08-28T06:35:01Z [INFO] [/core/main.go:100]: Start to sync quota data .....

.....

Aug 28 06:41:41 192.168.112.1 core[2686]: 2019-08-28T06:41:41Z [ERROR] [/core/api/quota/registry/registry.go:280]: pq: got 86139 parameters but PostgreSQL only supports 65535 parameters
Aug 28 06:41:41 192.168.112.1 core[2686]: 2019-08-28T06:41:41Z [ERROR] [/core/main.go:102]: Fail to sync quota data, pq: got 86139 parameters but PostgreSQL only supports 65535 parameters
Aug 28 06:41:41 192.168.112.1 core[2686]: 2019-08-28T06:41:41Z [FATAL] [/core/main.go:242]: quota migration error, pq: got 86139 parameters but PostgreSQL only supports 65535 parameters
```

We should introduce paging mechanism when populating the DB

Additionally after the core restarts the sync process triggered again, there's a different error:
```
Aug 28 06:51:07 192.168.112.1 core[2686]: 2019-08-28T06:51:07Z [INFO] [/core/api/quota/chart/chart.go:218]: API controller for chart repository server is successfully initialized
Aug 28 06:55:19 192.168.112.1 core[2686]: 2019-08-28T06:55:19Z [INFO] [/replication/registry/healthcheck.go:58]: Start regular health check for registries with interval 5m0s
Aug 28 06:55:44 192.168.112.1 core[2686]: 2019-08-28T06:55:44Z [ERROR] [/core/api/quota/registry/registry.go:280]: pq: duplicate key value violates unique constraint "unique_project_blob"
Aug 28 06:55:44 192.168.112.1 core[2686]: 2019-08-28T06:55:44Z [ERROR] [/core/main.go:102]: Fail to sync quota data, pq: duplicate key value violates unique constraint "unique_project_blob"
Aug 28 06:55:44 192.168.112.1 core[2686]: 2019-08-28T06:55:44Z [FATAL] [/core/main.go:242]: quota migration error, pq: duplicate key value violates unique constraint "unique_project_blob"
```

### PR DESCRIPTION
**Pull request by wy65701436**
1, fix #8858, add retry to ping backend service
2, fix #8859, split the blobs data when larger then 65535

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/common/dao/project_blob.go [hunk 1]
```diff
@@ -35,22 +35,43 @@ func AddBlobToProject(blobID, projectID int64) (int64, error) {
...
 // AddBlobsToProject ...
 func AddBlobsToProject(projectID int64, blobs ...*models.Blob) (int64, error) {
+	blobCounts := len(blobs)
 	if len(blobs) == 0 {
 		return 0, nil
 	}
 
-	now := time.Now()
+	// 65535 is the maximum support parameters, if larger than 65535, will divide them into chunks with size 10000
+	var dividedBlobs [][]*models.Blob
```
Comment (reasonerjt) [inline src/common/dao/project_blob.go]: I think even when the length of the `blobs` array is less then 65535, we should divide them into chunks, maybe 1024 or 4096 records in one chunk.

And this is a little trivial but seems you can do that in one loop.

File: src/core/api/quota/migrator.go [hunk 2]
```diff
@@ -111,6 +123,10 @@ func Sync(pm promgr.ProjectManager, populate bool) error {
 	return nil
 }
 
+func Check(f func() error) error {
+	return retry(10, 100*time.Millisecond, f)
```
Comment (reasonerjt) [inline src/core/api/quota/migrator.go]: first retry will happen in 0.1 second?

Comment (wy65701436) [reply src/core/api/quota/migrator.go]: no, please just see the retry func details.

File: src/common/dao/project_blob.go (hunk 1 above)
Comment (wy65701436) [reply src/common/dao/project_blob.go]: will update it to 1024

File: src/core/api/quota/migrator.go (hunk 2 above)
Comment (wy65701436) [reply src/core/api/quota/migrator.go]: the data of about ten times,

123.889705ms
341.617085ms
782.73897ms
2.152148225s
4.482828918s
12.274047382s
32.847927757s
1m21.819425406s
3m52.175972572s

File: src/core/api/quota/registry/registry.go [hunk 3]
```diff
@@ -277,6 +277,9 @@ func persistPB(projects []quota.ProjectInfo) error {
 		}
 		_, err = dao.AddBlobsToProject(pro.ProjectID, blobsOfPro...)
 		if err != nil {
+			if err == dao.ErrDupRows {
+				continue
```
Comment (reasonerjt) [inline src/core/api/quota/registry/registry.go]: log it?

File: src/common/dao/project_blob.go (hunk 1 above)
Comment (wy65701436) [reply src/common/dao/project_blob.go]: use the single insert instead

File: src/common/dao/project_blob.go [hunk 4]
```diff
@@ -34,6 +34,7 @@ func AddBlobToProject(blobID, projectID int64) (int64, error) {
 }
 
 // AddBlobsToProject ...
+// Note: pq has limitation on support parameters, the maximum length of blobs is 65535
```
Comment (heww) [inline src/common/dao/project_blob.go]: change `GetOrmer().InsertMulti(len(projectBlobs), projectBlobs)` to `GetOrmer().InsertMulti(1024, projectBlobs)`  (1024 can be other values) can make `AddBlobsToProject` to handle blobs more than 65535. 

`InsertMulti` in beego orm will split the values by bulk and insert each bulk values one by one.

File: src/core/api/quota/registry/registry.go [hunk 5]
```diff
@@ -275,11 +280,17 @@ func persistPB(projects []quota.ProjectInfo) error {
...
-			return err
+		for _, blobOfPro := range blobsOfPro {
+			_, err = dao.AddBlobToProject(blobOfPro.ID, pro.ProjectID)
+			if err != nil {
+				log.Error(err)
+				if err == dao.ErrDupRows {
+					continue
+				}
+				return err
+			}
```
Comment (heww) [inline src/core/api/quota/registry/registry.go]: Suggest to use AddBlobsToProject which will be faster than insert blob to project one by one.

**Pull request merged**